### ISSUE TITLE
[BUG] Memory increasing with ActivityOutputRecord

### ISSUE THREAD

**Issue by iio888** [elsa-workflows/elsa-core#6158]
## Description
Program memory will increase continuously when execute a activity with result in a loop.

## Steps to Reproduce
Here is my workflow to run.
```
internal class EndlessWorkflow : WorkflowBase
{
    protected override void Build(IWorkflowBuilder builder)
    {
        builder.Root = While.True(new If(c => true) 
        {
            Then = new Inline(c => Console.WriteLine(“loop”)
        };
    }
} 
```

## Actual Behavior
The number of `_records` in `ActivityOutputRegister` is increasing continuously. The memory is going to be 700M after 15 min.

## Environment
- **Elsa Package Version**:  3.2.3

**Comment by iio888** [comment]
I noticed that there is a similar issue #4900 which has been closed. But I think the core problem has not  been resolved. The real reason causes this issues is that the activity (`If` in my example)  is setting result continuously (call `Set` method) and the `ActivityOutputRecord` is created continuously at the same time.
In my example code, my workflow won’t exit. So `ActivityOutputRegister` won’t be disposed, too.

**Comment by sfmskywalker** [comment]
I think that makes sense, because for each activity execution, any of its output is captured. So the question is, what kind of controls should we offer to manage this? For example:

- Provide a threshold configuration that will discard older output.
- Disable collecting output altogether
- Flush output to the underlying store during commit states (which is part of a separate issue).

Thoughts?

**Comment by iio888** [comment]
I prefer the third advice. Disabling will also work if I don’t specify output target.

**Comment by shabbirlathsaheb-gep** [comment]
has anyone implemented recommended fixes? @iio888 @sfmskywalker . We are facing the same issue.

**Comment by sfmskywalker** [comment]
Yes, Elsa 3.4 (in preview) has a new feature called **Commit Strategies**, which allows you to control at which points in the workflow to commit the current state - which will also flush the activity execution records.

For scenarios like the EndlessWorkflow, you **could** commit state after every iteration, but this would slow down the performance of this workflow. A smarter way would be to apply the `PeriodicWorkflowStrategy` to the workflow, which periodically commits the workflow state, say, every 10 seconds. This way, you ensure that built-up records are flushed to the DB without also causing excessive DB traffic which would slow down the workflow.

To enable the Commit Strategies feature:

```csharp
elsa.UseWorkflows(workflows =>
{
    workflows.UseCommitStrategies(strategies =>
    {
        strategies.AddStandardStrategies();
        strategies.Add("Every 10 seconds", new PeriodicWorkflowStrategy(TimeSpan.FromSeconds(10)));
    });
})
```

You can now select the desired strategy from the workflow properties:

![Image](https://github.com/user-attachments/assets/c903b9ec-4a3a-4720-84ff-d2da5da87a46)

Although I haven't tried, but instead of doing the Periodic commit strategy at the workflow level, you might try implementing a custom `IActivityCommitStrategy` that implements the periodic behavior similar to `PeriodicWorkflowStrategy`. You would apply this strategy to e.g. the While activity in the EndlessWorkflow.
